File: python/challenge-2025-09-11.py
def reverse_sentence(sentence):
    word_list = sentence.split()
    word_list.reverse()
    result = " ".join(word_list)
    return result

def test_io(input):
    output = reverse_sentence(input)
    print(f" - input:   {input}")
    print(f" - output:  {output}")

def run_tests(inputs):
    print("Running tests...")
    for i, input in enumerate(inputs, start=1):
        print(f"Test {i}:")
        test_io(input)
    print("Testing complete!")

def main():
    inputs = [
        "world hello",
        "push commit git",
        "npm  install   apt    sudo",
        "import    default   function  export"
    ]
    run_tests(inputs)

if __name__ == "__main__":
    main()

File: javascript/challenge-2025-09-11.js
function reverseSentence(sentence) {
  const word_list = sentence.split(/\s+/);
  word_list.reverse();
  let result = word_list.join(" ");
  return result;
}

function test_io(input) {
  let output = reverseSentence(input);
  console.log(" - input:", input);
  console.log(" - output:", output);
}

function run_tests(inputs) {
  console.log("Running tests...");
  for (let i = 0; i < inputs.length; i++) {
    let n = i + 1;
    console.log("Test %d:", n);
    test_io(inputs[i]);
  }
  console.log("Testing complete!");
}

function main() {
  const inputs = [
    "world hello",
    "push commit git",
    "npm  install   apt    sudo",
    "import    default   function  export"
  ];
  run_tests(inputs);
}

if (require.main == module) {
  main();
}
